src/catalog/src/expr_cache.rs

```diff
@@ -78,38 +74,60 @@ impl DurableCacheCodec for ExpressionCodec {
     ) {
         (
             RelationDesc::builder()
-                .with_column("data", ScalarType::Jsonb.nullable(false))
+                .with_column("key", ScalarType::Jsonb.nullable(false))
+                .with_column("val", ScalarType::Jsonb.nullable(false))
                 .finish(),
             UnitSchema::default(),
         )
     }
 
     fn encode(key: &Self::Key, val: &Self::Val) -> (Self::KeyCodec, Self::ValCodec) {
-        let codec_key = CodecKey {
-            cache_key: Cow::Borrowed(key),
-            expressions: Cow::Borrowed(val),
-        };
-        let serde_value = serde_json::to_value(codec_key).expect("valid json");
-        let jsonb =
-            Jsonb::from_serde_json(serde_value).expect("contained integers should fit in f64");
-        let row = jsonb.into_row();
+        let serde_key = to_value(key);
+        let mut row = Row::default();
+        let mut packer = row.packer();
+        JsonbPacker::new(&mut packer)
+            .pack_serde_json(serde_key)
+            .expect("contained integers should fit in f64");
+        JsonbPacker::new(&mut packer)
+            .pack_serde_json(val.clone())
+            .expect("contained integers should fit in f64");
```

Comment:
Spoke with @danhhz offline, these `expect`s are not necessarily true. If some plan has a constant that is a large int, it may not fit in `f64`. I'll have to think about this one.

Reply:
@danhhz the most recent commit has this switched to a string.